modin/core/dataframe/pandas/partitioning/partition_manager.py

```diff
@@ -1305,3 +1305,57 @@ def rebalance_partitions(cls, partitions):
             The same 2-d array.
         """
         return partitions
+
+    @classmethod
+    def shuffle_partitions(cls, partitions, sample_func, pivot_func, split_func):
+        """
+        Return shuffled partitions.
+
+        Parameters
+        ----------
+        partitions : np.ndarray
+            The 2-d array of partitions to shuffle.
+        sample_func : Callable(pandas.DataFrame) -> Any
+            Function to sample partitions. Output of this function will be passed to `pivot_func`
+            to determine pivots.
+        pivot_func : Callable(List[Any]) -> Any
+            Function to determine pivots from partitions. The pivots determined by this function
+            will be passed to `split_func` to split each partition.
+        split_func : Callable(pandas.DataFrame, Any) -> *List[pandas.Dataframe]
+            Function that splits a partition based off of the pivots provided. Should return an
+            unpacked list of pandas Dataframes.
+
+        Returns
+        -------
+        np.ndarray
+            A list of row-partitions that have been shuffled.
+        """
+        # Convert our partitions into row partitions
+        row_partitions = cls.row_partitions(partitions)
+        # Sample each partition
+        samples = [
+            partition.apply(sample_func, _impure=True) for partition in row_partitions
+        ]
+        # Get each sample to pass in to the pivot function
+        samples = [row.get() for frame in samples for row in frame]
+        pivots = pivot_func(samples)
+        # Get each ``partition_cls`` wrapped by the ``axis_partition_cls`` (should be 1 per axis
+        # partition since  we set ``num_splits`` to 1 earlier).
+        row_partitions = [
+            partition.list_of_partitions_to_combine[0] for partition in row_partitions
+        ]
+        # Gather together all of the sub-partitions
+        split_row_partitions = np.swapaxes(
+            np.array(
+                [
+                    partition.split(split_func, len(pivots), pivots)
+                    for partition in row_partitions
+                ]
+            ),
+            0,
```

Comment:
nit: add the keywords `axis1=0` and `axis2=1` here. I think it's clearer and prevents us from having to look up the docstring for `np.swapaxes`.